mantis/acquisition/acq_engine.py

```diff
@@ -793,59 +809,60 @@ def acquire_ls_defocus_stack(
 
         # Set config
         if config_name is not None:
-            mmc.set_config(config_group, config_name)
-            mmc.wait_for_config(config_group, config_name)
+            mmc.setConfig(config_group, config_name)
+            mmc.waitForConfig(config_group, config_name)
 
         # Set exposure time
         if exposure_time is not None:
-            mmc.set_exposure(exposure_time)
+            mmc.setExposure(exposure_time)
 
         # Open shutter
         auto_shutter_state, shutter_state = microscope_operations.get_shutter_state(mmc)
         microscope_operations.open_shutter(mmc)
 
         # get galvo starting position
-        p0 = mmc.get_position(galvo)
+        p0 = mmc.getPosition(galvo)
 
         # set camera to internal trigger
         # TODO: do this properly, context manager?
         microscope_operations.set_property(
             mmc, 'Prime BSI Express', 'TriggerMode', 'Internal Trigger'
         )
-        if use_pycromanager:
+        if use_pymmcore_plus:
             tempdir = TemporaryDirectory()
-            focus_stage = mmc.get_property('Core', 'Focus')
+            focus_stage = mmc.getProperty('Core', 'Focus')
             microscope_operations.set_property(mmc, 'Core', 'Focus', z_stage)
 
         # acquire stacks at different galvo positions
         for p_idx, p in enumerate(galvo_range):
             # set galvo position
-            mmc.set_position(galvo, p0 + p)
+            mmc.setPosition(galvo, p0 + p)
 
             # acquire defocus stack
-            if use_pycromanager:
-                mmc.set_position(z_stage, z_range[0])  # prep o3 stage
-                with Acquisition(
-                    tempdir.name, f'ls_refocus_p{p_idx}', port=LS_ZMQ_PORT, show_display=False
-                ) as acq:
-                    acq.acquire(
-                        multi_d_acquisition_events(
-                            z_start=z_range[0],
-                            z_end=z_range[-1],
-                            z_step=z_range[1] - z_range[0],
-                        ),
-                    )
-                ds = acq.get_dataset()
-                data.append(np.asarray(ds.as_array()))
-                ds.close()
-                mmc.set_position(z_stage, z_range[len(z_range) // 2])  # reset o3 stage
+            if use_pymmcore_plus:
+                mmc.setPosition(z_stage, z_range[0])  # prep o3 stage
+
+                # acquire defocus stack
+                mda = useq.MDASequence(z_plan=useq.ZAbsolutePositions(z_range))
+
+                # append data as its acquired.
+                def append_data(img: np.ndarray, event: useq.MDAEvent):
+                    data.append(img)
+                mmc.mda.events.frameReady.connect(append_data)
+
+                # run the acquisition, and wait for it to finish
+                mmc.run_mda(mda)
+                mmc.waitForAcquisition()
```

Comment:
quick note:  `mmc.run_mda` is a [convenience method](https://github.com/pymmcore-plus/pymmcore-plus/blob/8742aa5435af119ef2b1cf7630ec55366d59b5a9/src/pymmcore_plus/core/_mmcore_plus.py#L1499-L1502) for starting a thread and calling `mmc.mda.run` in that thread.  If you just want to block immediately and wait, you can call `mmc.mda.run(mda)` directly and avoid the thread altogether.